compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaObjectTypeSymbol.java

```diff
@@ -147,7 +147,7 @@ public BallerinaObjectTypeSymbol(CompilerContext context, ModuleID moduleID, BOb
     public String signature() {
         StringBuilder signature = new StringBuilder();
         StringJoiner qualifierJoiner = new StringJoiner(" ");
-        StringJoiner fieldJoiner = new StringJoiner(";");
+        StringJoiner fieldJoiner = new StringJoiner("");
```

Comment:
Not requesting to change, but just wondering why is this changed, this string joiner looks like a string builder.

Reply:
`new StringJoiner(";", "", ";")`

Reply:
This is the first thing I tried. The problem is prefix gets added to an empty joiner too. Can add this and only add the joiner conditionally to the signature.